src/junit.rs

```diff
@@ -178,7 +191,10 @@ pub fn use_reader(
                     in_failure = true;
                 }
                 b"testsuite" => {
-                    testsuite_names.push(get_attribute(&e, "name")?);
+                    testsuite_names.push(
+                        get_attribute(&e, "name")?
+                            .map(|s| ValidatedString::from_string(s).unwrap()),
```

Comment:
```suggestion
                        get_attribute(&e, "name")
                            .and_then(ValidatedString::from_string),
```

The `and_then` combinator can chain multiple `Result`-returning calls together which is quite handy.

Either way, we should avoid the `unwrap` and rather propagate that error.